src/main/java/seedu/task/logic/parser/AddParser.java

```diff
@@ -45,24 +52,22 @@ public Command prepare(String args){
         final Matcher taskMatcher = TASK_DATA_ARGS_FORMAT.matcher(args.trim());
         final Matcher eventMatcher = EVENT_DATA_ARGS_FORMAT.matcher(args.trim());
         
+        ArgumentTokenizer argsTokenizer = new ArgumentTokenizer(descriptionPrefix, deadlinePrefix, durationPrefix);
```

Comment:
Actually, this is just a thought, but since we are using the Tokenizer as a form of capturing the specific groups of data. Can we use it to authenticate the args instead of the Matcher class? Hence, we no longer need to regex and Matcher class.